service/lib/agama/dbus/storage/volume_conversion/from_dbus.rb

```diff
@@ -0,0 +1,147 @@
+require "y2storage/disk_size"
+require "y2storage/filesystems/type"
+
+module Agama
+  module DBus
+    module Storage
+      module VolumeConversion
+        # Volume conversion from D-Bus format.
+        class FromDBus
+          # @param dbus_volume [Hash]
+          # @param config [Agama::Config]
+          def initialize(dbus_volume, config:)
+            @dbus_volume = dbus_volume
+            @config = config
+          end
+
+          # Performs the conversion from D-Bus format.
+          #
+          # @return [Agama::Storage::Volume]
+          def convert
+            builder = Agama::Storage::VolumeTemplatesBuilder.new_from_config(config)
+            volume = builder.for(dbus_volume["MountPath"] || "")
+
+            volume.tap do |target|
+              dbus_volume.each do |dbus_property, dbus_value|
+                converter = CONVERSIONS[dbus_property]
+                # FIXME: likely ignoring the wrong attribute is not the best
+                next unless converter
+
+                send(converter, target, dbus_value)
+              end
+            end
+          end
+
+        private
+
+          # @return [Hash]
+          attr_reader :dbus_volume
+
+          # @return [Agama::Config]
+          attr_reader :config
+
+          # D-Bus attributes and their converters.
+          CONVERSIONS = {
+            "MountPath"    => :mount_path_conversion,
+            "MountOptions" => :mount_options_conversion,
+            "TargetDevice" => :target_device_conversion,
+            "TargetVG"     => :target_vg_conversion,
+            "FsType"       => :fs_type_conversion,
+            "MinSize"      => :min_size_conversion,
+            "MaxSize"      => :max_size_conversion,
+            "AutoSize"     => :auto_size_conversion,
+            "Snapshots"    => :snapshots_conversion
+          }.freeze
+          private_constant :CONVERSIONS
+
+          # @param target [Agama::Storage::Volume]
+          # @param value [String]
+          def mount_path_conversion(target, value)
+            target.mount_path = value
+          end
+
+          # @param target [Agama::Storage::Volume]
+          # @param value [Array<String>]
+          def mount_options_conversion(target, value)
+            target.mount_options = value
+          end
+
+          # @param target [Agama::Storage::Volume]
+          # @param value [String]
+          def target_device_conversion(target, value)
+            target.device = value
+          end
+
+          # @param target [Agama::Storage::Volume]
+          # @param value [String]
+          def target_vg_conversion(target, value)
+            target.separate_vg_name = value
+          end
+
+          # @param target [Agama::Storage::Volume]
+          # @param value [String]
+          def fs_type_conversion(target, value)
+            fs_type = target.outline.filesystems.find do |type|
+              type.to_human_string.downcase == value.downcase
```

Comment:
small efficiency note: `value.downcase` can be computed outside of loop.
another option is `type.to_human_string.casecmp?(value)`